### DISCUSSION TITLE
CSS for FormBuilder

### DISCUSSION THREAD

**Discussion by minh-tranchi** [optimajet/formengine#26] [General]
I'm having trouble sending email with the HTML getting from the FormViewer
![image](https://github.com/optimajet/formengine/assets/141224728/cfe01bfe-ba7f-4d25-8d2b-1f50bc1de692)
![image](https://github.com/optimajet/formengine/assets/141224728/17f0b0ba-c2dd-4662-bd4a-23c375775138)
Knowing that Gmail will auto remove class in HTML so the CSS that was compiled to the class will not work in the Gmail. In the image, there are flex and font size difference because I use inline style for it.

Is there anyway to get the Style from the FormBuilder or from the compiled class?

**Comment by optimajet** [comment]
Currently, all styles are compiled into a CSS class and this class is added to the component's `className` (https://formengine.io/documentation/component-styling#styling-a-simple-component) property. FormEngine uses the `cx` (emotion) function to compile styles into a CSS class. There is no easy way to get these styles.

Seems like a mode to compile styles to the inline styles of the component would be useful for such cases?

**Reply by minh-tranchi** [reply]
Yes that would be greate for such cases. Then for now is there anyway i can mapping style props in the json to inline style for my  custom components ?  @optimajet

**Reply by optimajet** [reply]
@minh-tranchi Currently, there are no such methods available. We are investigating possible solutions for this case. Could you please describe in more detail how you embed FormEngine into an email? Maybe you have an example of an email?

**Reply by minh-tranchi** [reply]
> e no such methods available. We are investigating possible solutions for this case. Could you please describe in more detail how you embed FormEngine into an email? Maybe you have an example of an email?

Currently, I'm using useRef to get HTML from the FormViewer and embed it to the email
![image](https://github.com/optimajet/formengine/assets/141224728/d14c01d8-f7c9-4943-b866-3f7d3528bdf7)
This is the HTML format that I get from the FormViewer using inline style

**Reply by optimajet** [reply]
Is it correct that you don't use JavaScript in the form (i.e. you don't use the FormViewer itself in the email)? Do you basically need to be able to get HTML with inline styles?

**Reply by minh-tranchi** [reply]
Yes, I don't use JS in the form. I really need to get the HTML with inline style so that the user can receiver a better email format.

**Reply by optimajet** [reply]
> Yes, I don't use JS in the form. I really need to get the HTML with inline style so that the user can receiver a better email format.

We added the task of applying styles directly to the component to our backlog. But this task is not a priority at the moment.

**Comment by minh-tranchi** [comment]
![image](https://github.com/optimajet/formengine/assets/141224728/34d0912c-b03d-4c86-8cd1-fa1114deb423)
Is there any way to disable dark mode and always show light mode? I hide the header for implementation so I want to set it to light mode by default @optimajet

**Reply by optimajet** [reply]
> Is there any way to disable dark mode and always show light mode? I hide the header for implementation so I want to set it to light mode by default

The easiest way to do this is to store the `theme` (light) in localStorage:

<img width="1916" alt="image" src="https://github.com/optimajet/formengine/assets/10303570/e355862c-6967-48ca-90ad-2bdcd6072a9d">

**Reply by sergeythrees** [reply]
Hi there, we have update:
https://formengine.io/documentation/theme-manipulation
